### PR TITLE
Add more Marshal exception tests

### PR DESCRIPTION
**Pull request by hughbe**
Separated from https://github.com/dotnet/corefx/pull/31057

### CODE REVIEW

File: src/System.Runtime.InteropServices/tests/System/Runtime/InteropServices/Marshal/ThrowExceptionForHRTests.cs [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+                }
+                else
+                {
+                    Assert.Equal("mscorlib", ex.Source);
+                }
+                Assert.Contains(nameof(ThrowExceptionForHR_ErrorInfo_ReturnsValidException), ex.StackTrace);
+                Assert.Contains(nameof(Marshal.ThrowExceptionForHR), ex.TargetSite.Name);
+            }
+
+            Assert.True(calledCatch);
```
Comment (AaronRobinsonMSFT) [inline src/System.Runtime.InteropServices/tests/System/Runtime/InteropServices/Marshal/ThrowExceptionForHRTests.cs]: General practice here is to put an `Assert.Fail(...);` below the `throw` statement. Then the bool can be removed entirely.

Comment (AaronRobinsonMSFT) [reply src/System.Runtime.InteropServices/tests/System/Runtime/InteropServices/Marshal/ThrowExceptionForHRTests.cs]: Throughout.

Comment (hughbe) [reply src/System.Runtime.InteropServices/tests/System/Runtime/InteropServices/Marshal/ThrowExceptionForHRTests.cs]: Not sure such a method exists
> System/Runtime/InteropServices/Marshal/ThrowExceptionForHRTests.cs(22,24): error CS0117: 'Assert' does not contain a definition for 'Fail' [/Users/hugh/Documents/GitHub/corefx/src/System.Runtime.InteropServices/tests/System.Runtime.InteropServices.Tests.csproj]
System/Runtime/InteropServices/Marshal/ThrowExceptionForHRTests.cs(62,24): error CS0117: 'Assert' does not contain a definition for 'Fail' [/Users/hugh/Documents/GitHub/corefx/src/System.Runtime.InteropServices/tests/System.Runtime.InteropServices.Tests.csproj]

Comment (AaronRobinsonMSFT) [reply src/System.Runtime.InteropServices/tests/System/Runtime/InteropServices/Marshal/ThrowExceptionForHRTests.cs]: Sad. This approach is fine then. Thanks.

**Comment by danmosemsft** [comment]
Thanks!

**Pull request merged**